GH-3734: Increase default max stack height

mpapierski opened the pull request:
Closes #3734

Increase default max stack height.

This should enable the usage of certain 3rd party crates that are quite expensive in terms of stack height cost.

Commit:
GH-3734: Increase default max stack height

This should enable usage of certain 3rd party crates that are quite expensive
in terms of stack height cost.

rafal-ch reviewed (approved):
LGTM :+1:

Fraser999 reviewed (commented):
We should probably mention this change in the EE changelog.

Commit:
Added changelog entry for this change

mpapierski commented:
bors r+

mpapierski commented:
bors r+

mpapierski commented:
bors r+

Commit:
Fix tests

These tests were previously exercising parameter limit, but were changed
as part of lowering stack height limits.

mpapierski commented:
bors r+

mpapierski commented:
bors r+
